pkg/vminetworkscontroller/vmi_controller.go

```diff
@@ -193,8 +190,27 @@ func (r *VirtualMachineInstanceReconciler) vmiNetworksClaimingIPAM(
 			if nadConfig.AllowPersistentIPs {
 				vmiNets[net.Name] = nadConfig.Name
 			}
+		} else if net.Pod != nil {
```

Comment:
can we turn this around ? i.e. handle the pod network **first** ?

Reply:
consider please adding the new logic under this condition in a function

Reply:
nah good point; if you can turn this into an helper, it would for sure be better. 

It is not a must from my perspective, can be tackled as a follow-up IMHO.

Reply:
I will do a pair of functions to make is symmetric, one for for secondary and other for primary UDNs